runtime/src/qio/qio.c

```diff
@@ -2328,6 +2328,9 @@ qioerr _buffered_read_atleast(qio_channel_t* ch, int64_t amt)
     left -= num_read;
     qbuffer_iter_advance(&ch->buf, &read_start, num_read);
 
+    // Ignore interrupted system call, just keep reading.
+    if( err && qio_err_to_int(err) == EINTR ) err = 0;
+
```

Comment:
Some systems will return from a system call with EINTR if a signal was received and it needed to run the signal handler. The signal handler for control-c is what would kill the process. If the signal was handled (and didn't kill the process), we might like to be able to continue.

I personally think EINTR is just a *nix wart that we have to work around. I don't think it is meaningful.